topomodelx/nn/hypergraph/dhgcn_layer.py

```diff
@@ -169,25 +169,27 @@ def get_dynamic_topology(self, x_0_features):
         """
         device = x_0_features.device
         n_nodes = x_0_features.size(0)
-        batch = torch.zeros(x_0_features.size(0), dtype=torch.long, device=device)
+        batch = torch.zeros(x_0_features.size(0), dtype=torch.long, device="cpu")
```

Comment:
Why hardcoding the device as the cpu? Users might want to process everything on the device of their choice. Consider changing.

Reply:
I have checked and knn and k-means doesn't work with GPU